osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/mapping/GridMeasurementPublishedEventConverter.java

```diff
@@ -38,28 +44,89 @@
 
     private static final int SIMPLE_END_INDEX = 10;
 
+    private ObjectMapper objectMapper = new ObjectMapper();
+
     @Override
     public GridMeasurementPublishedEvent convert(final String source,
             final Type<? extends GridMeasurementPublishedEvent> destinationType, final MappingContext mappingContext) {
 
         StringArrayToAnalogList stringArrayToAnalogList = null;
-        final String[] values = source.split(";");
-        if (values.length == SIMPLE_END_INDEX) {
-            stringArrayToAnalogList = new SimpleStringToAnalogList();
-        } else if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
-            stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
-        } else {
-            LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+        LOGGER.info("Source string: {}", source);
+
+        try {
+            final Payload payload = this.objectMapper.readValue(source, Payload[].class)[0];
+
+            final String[] values = (payload.gisnr + ", " + String.join(", ", payload.data)).split(", ");
+            LOGGER.info("Values length: {} and values: {}", values.length, Arrays.toString(values));
+
+            if (values.length == SIMPLE_END_INDEX) {
+                stringArrayToAnalogList = new SimpleStringToAnalogList();
+            } else if (values.length == LsPeakShavingMeasurementType.getNumberOfElements() + 1) {
+                stringArrayToAnalogList = new LsMeasurementMessageToAnalogList();
+            } else {
+                LOGGER.error("String '{}' does not have the expected amount of fields, abandoning conversion", source);
+                return null;
+            }
+
+            final String eanCode = values[0];
+            final PowerSystemResource powerSystemResource = new PowerSystemResource(eanCode,
+                    UUID.randomUUID().toString(), new ArrayList<Name>());
+            final long createdDateTime = System.currentTimeMillis();
+            return new GridMeasurementPublishedEvent(createdDateTime, eanCode, UUID.randomUUID().toString(),
+                    "GridMeasurementPublishedEvent", new ArrayList<Name>(), powerSystemResource,
+                    stringArrayToAnalogList.convertToAnalogList(values));
+        } catch (final JsonMappingException e) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AS1166)